SDK/AppCenter/Microsoft.AppCenter.WindowsDesktop/Utils/DefaultApplicationSettings.cs

```diff
@@ -68,6 +68,7 @@ private void SaveValue(string key, string value)
         {
             lock (configLock)
             {
+                configuration = OpenConfiguration();
```

Comment:
Wait are we trying to open the file for every single property change?
Do we have issues with read?

Reply:
Also, do we have to release the file in some way?  I would expect that this would prevent writing saving values in the second application at all

Reply:
We will parking lot this as this can cause a lot of side effects with inconsistent states between saved. Like A disable only crashes, B disables only Push, what does B save?

Reply:
We have several (same) applications that make changes in the same file. As I understand, the config file was opened for writing only when the 'Configuration.Save' method works.
So we have the following situation. The first app changed the config file. In this case, the second app can't save changes without configuration refreshing
(we see the exception and that is why I have added refreshing current configuration from file). In this case, the configuration should be refreshed each time when we save changes (before 'Save' method called). 
Because when the first app made changes in the file then the second app should refresh configuration. After that, the second app saves other changes to the file and the first app configuration should be refreshed or the same exception will appear in the first app.
If the first app disables only 'Crashes' and after that, the second app disables only 'Push' then the 'Crashes' and the 'Push' both will be disabled in the second application.
Maybe we should decide what solution is better:
1. All app instances could write the app config but in this case, this is a problem to check that all app instances have the right configuration (maybe this is a problem of the application authors).
2. Only one application instance could write to our configuration file and when the second app tries to write some changes then we write some logs and ignore these changes.
Maybe there are other ways to solve this problem.

Reply:
We decided to not reopen the file every time.
See our internal conversation thread for details.